### PR TITLE
Documentation review

### LINKED ISSUE #117: Document breaking BC change in config introduced in the last release
With #101 we changed the behavior of the configuration reader. 
We should make clear what is changed and provide a suggestion to migrate to the newer release.

### PR DESCRIPTION
**Pull request by ilario-pierbattista**
Resolves #117

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -27,17 +37,19 @@ Then, enable the bundle on the `AppKernel` class:
...
 ```php
 // app/AppKernel.php
-
-public function registerBundles()
-{
-    return [
-        // ...
-        new Facile\MongoDbBundle\FacileMongoDbBundle(),
-    ];
+class AppKernel extends Kernel {
```
Comment (Jean85) [inline README.MD:40]: This is not useful with Flex, where we have a recipe: https://github.com/symfony/recipes-contrib/tree/master/facile-it/mongodb-bundle/0.6

File: README.MD [hunk 2]
```diff
@@ -95,64 +123,60 @@ $this->get('mongo.connection.{connectionName}'); // [test_db, other_db, test_db_
 
 To manipulate the database, please read the official [documentation](http://mongodb.github.io/mongo-php-library/classes/database/)
 
-### Fixtures
+## Query Profiling
+
+On dev environment all queries executed by the library MongoDB\Collection class are profiled and showed inside the symfony web profiler.
```
Comment (Jean85) [inline README.MD:128]:
```suggestion
On dev environment all queries executed by the library MongoDB\Collection class are profiled and showed inside the Symfony web profiler.
```

File: README.MD [hunk 3]
```diff
@@ -95,64 +123,60 @@ $this->get('mongo.connection.{connectionName}'); // [test_db, other_db, test_db_
...
+
+[![Profiler Toolbar](https://github.com/facile-it/mongodb-bundle/blob/master/docs/img/profiler_toolbar.png)](https://github.com/facile-it/mongodb-bundle/blob/master/docs/img/profiler_toolbar.png)
+
+[![Profiler Panel](https://github.com/facile-it/mongodb-bundle/blob/master/docs/img/profiler_panel.png)](https://github.com/facile-it/mongodb-bundle/blob/master/docs/img/profiler_panel.png)
+
+## Fixtures
 This bundle supports doctrine style fixtures.
-To create one you will need to add a php class in your bundle's `/DataFixtures/Mongo` directory. 
+To create one you will need to add a php class in your bundle's `/DataFixtures/Mongo` directory.
```
Comment (Jean85) [inline README.MD:136]: I didn't have to do this in my project... I do `bin/console mongodb:fixtures:load tests/Fixtures/Mongo`

File: README.MD (hunk 1 above)
Comment (ilario-pierbattista) [reply README.MD:40]: Btw, that recipe should be updated with the explicit `authSource: '%env(MONGODB_DB)%'`, shouldn't it?

Comment (Jean85) [reply README.MD:40]: If you think it's a sensible default, yes.
You can duplicate the recipe and add a version constraint to 1.10 if you want.

File: README.MD (hunk 3 above)
Comment (ilario-pierbattista) [reply README.MD]: @Jean85 Thanks. Fixed and restructured the whole section: https://github.com/facile-it/mongodb-bundle/tree/2105_%23117_sync_doc#fixtures
Is it clear?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+Here is the configuration reference:
+
+```yaml
+mongo_db_bundle:
+
+  data_collection: true # set to false to disable data collection
+
+    # clients section, here you can define connection to different servers or with different credentials
+    clients:
```
Comment (Jean85) [inline README.md:69]: Indentation seems wrong here

**Review by Jean85 — CHANGES_REQUESTED** [review]
Apart from an indentation error in the example, it looks good! :+1: 

It's very long though... I would suggest splitting it into:
 * Readme.md
 * Upgrade.md
 * Contributing.md
 * Documentation.md

File: docs/Contributing.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+### Tests and Quality Tools
+
+This project uses `phpunit` to run its test suite.
+Also, it uses `phpstan` for static code analysis and `php-cs-fixer` for code style.
+
+It's common to run the following in sequence.
+
+    make test
+    make phpstan
+    make cs-fix
```
Comment (Jean85) [inline docs/Contributing.md:80]: I would suggest using the reverse order, since CS could theoretically impact PHPStan

**Review by salvatorecordiano — APPROVED** [review]

**Review by Jean85 — APPROVED** [review]

**Pull request merged**